Jahsie/cards3-template.js

```diff
@@ -0,0 +1,187 @@
+    return NaN;
+}
+    cardRank = Math.floor(this.id/4) + 1;
+    return cardRank;
+};
+
+makeCard.suit = function() { // --> 1..4, NaN
+    id = this.id;
+    if (!(Number.isInteger(id)) || id > 51 || id < 0) {
+    return NaN;
+}
+    suitLabel = makeCard.suitName[id%4];
+    return suitLabel;
+};
+   
+makeCard.color = function() { // -->"red,"black",NaN
+    id = this.id;
+    if (!(Number.isInteger(id)) || id > 51 || id < 0) {
+    return NaN;
+}
+    if (this.suit(id) < 3) {
+        cardColor = "red";
+    } else {
+        cardColor = "black";
+        }
+    return cardColor;
+};
+
+makeCard.cardName = function() { //--> string, NaN
+    id = this.id;
+    // This method can't have the key 'name' within the makeCard function,
+    // but instance objects can store a reference to it called 'name'
+    if (!(Number.isInteger(id)) || id > 51 || id < 0) {
+    return NaN;
+}
+    theCardName = makeCard.suitTitle[(this.suit()) -1];
+    nameOfCard= (makeCard.rankName[(this.rank()) - 1] + " of " + theCardName);
+    return nameOfCard;
+};
+
+
+var card12 = makeCard(12);
+// card12.rank();
+// card12.suit();
+// card12.color();
+// card12.cardName();
+
+//-----------------------
+// Methods to be called through factory only:
+//-----------------------
+
+makeCard.isCard = function(obj) { // --> true,false
+    // if (thing > 51 || thing < 0 || !(Number.isInteger(thing)) || !(typeof thing == 'number')) {
+    //     return false;
+    // }
+    // else return true;
+    var theTester = obj;
+    // if (!(theTester.rank typeof == 'function')) {
+    //     return false;
+    // } else { 
+    //     var theTesterRank = theTester.rank();
+        if (!(/*13 >= theTesterRank >= 0|| */51 >= theTester.id >= 0) || (!(theTester.color()))) {
+        return false;
+    } else {
+        return true;
+    }
+    
+};
+
+//---------------------
+// Additional factory properties
+//---------------------
+
+ //<-- instead, generate array of 52 card instances
+
+makeCard.fullSet = [];
+
+makeCard.makeFullSet = function() {
+    for (i = 0; i < 52; i++) {
+        //var theNewCard = "card" + i; 
+        //theNewCard = makeCard(i);
+        //var theNewCard = makeCard(i);
+        //makeCard.fullSet[i] = theNewCard;
+        //I based this of the work we did last week where we were adding new people objects to the people.index object.
+        makeCard.fullSet.push(makeCard(i));
+        //console.log(makeCard(i));
+    }
+    console.log("For this problem I spent a long time trying to give each card a name so that when I check makeCard.fullset I would see a list of objects named card1, card2 etc. After many attempts (see the stuff I commented out) I came to the conclusion that this isn't possible. Am I right in my conclusion? It seems that an array of objects will always look like this: array = [object, object]. Even if the objects have names they will still appear as 'object' when viewed in the array.");
```

Comment:
That's a good thought. Ultimately, how that gets displayed comes down to what decisions the browser makes, so there's not a lot of control you have over it. You might consider using a function like this: `function displayNames () { deckOfCards.array.forEach(function (x) {console.log(x.name());}); }` to show you what's going on in there.